src/styles/main.scss

```diff
@@ -1,3 +1,65 @@
+@import "nornalize";
+@import "variables";
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  gap: $gapSize;
+  box-sizing: border-box;
+  flex-wrap: wrap;
+  width: $calcWidth * 7 + $paddingCal * 2;
+  padding: $paddingCal;
+
+  $days:
+    ("mon": 0,
+    "tue":$calcWidth,
+    "wed":$calcWidth * 2,
+    "thu":$calcWidth * 3,
+    "fri": $calcWidth * 4,
+    "sat":$calcWidth * 5,
+    "sun": $calcWidth * 6);
+
+  @each $day, $margin-left in $days {
+    &--start-day--#{$day} {
+      .calendar__day:first-child {
+        margin-left: $margin-left;
+      }
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} {
+      .calendar__day:nth-child(-n+#{$i}) {
+        display: block;
+      }
+    }
+  }
+
+  &__day {
+    display: none;
+    position: relative;
+    width: $daySize;
+    height: $daySize;
+    background-color: #eee;
+    border: $borderSize solid #000;
+    cursor: pointer;
+    transition: background-color 0.5s, margin 0.5s;
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
```

Comment:
add all these styles outside the loop, leave in the loop styles that depend only on the iteration of the loop

Reply:
When I take these styles out of the loop, the program breaks

Reply:
Have you tried adding these styles to `&__day::before` selector?